### PR TITLE
Update mxo 1.2.1

### PR DESCRIPTION
**Pull request by chpypl**
Thank you for your contribution to Braintree. 

> Before submitting this PR, note that we cannot accept language translation PRs. We support the same languages that are supported by PayPal, and have a dedicated localization team to provide the translations. If there is an error in a specific translation, you may open an issue and we will escalate it to the localization team.

### Summary of changes

 - Update mxo 1.2.1

 ### Checklist

 - [X] Added a changelog entry

-

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Braintree Android SDK Release Notes
 
+## unreleased
+
+* PayPalNativeCheckout
+  * Bump native-checkout version to release `1.2.1`
```
Comment (jaxdesmarais) [inline CHANGELOG.md:6]: We should add a line to the changelog that this fixes the Google Play compliance issue so it's a bit easier to find what version of MXO resolved the issue.

**Review by sarahkoop — APPROVED** [review]

**Review by sshropshire — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (chpypl) [reply CHANGELOG.md:6]: This is what we put in the release notes for direct integration. Do we want to copy and paste it? https://github.com/paypal/android-checkout-sdk/releases/tag/v1.2.1

Comment (hollabaq86) [reply CHANGELOG.md:6]: we don't need that entire paragraph, maybe we just hyper link to android-checkout-sdk's release notes for folks to go read for themselves the updates?

Comment (chpypl) [inline CHANGELOG.md:6]:
```suggestion
  * Bump native-checkout version to release `1.2.1`
  * Upgraded the data-collector SDK to version 3.21.0 which made updates to Device Data collection related to Google Play's User Data Policy. For more info read the [release notes](https://github.com/paypal/android-checkout-sdk/releases/tag/v1.2.1)
```

Comment (chpypl) [reply CHANGELOG.md:6]: @hollabaq86 @jaxdesmarais how about this?

Comment (jaxdesmarais) [reply CHANGELOG.md:6]: That sounds good to me!

**Pull request merged**